File: TreasurersApp/Models/AddressTypeActionResult.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace TreasurersApp.Models
{
    public class AddressTypeActionResult : WebResult<AddressType>
    {
        public AddressTypeActionResult(bool success, List<string> statusMessages, AddressType addressType)
            :base(success, statusMessages, addressType)
        {
        }
    }
}

File: TreasurersApp/ClientApp/src/app/models/AddressTypeActionResult.ts
  import { AddressType } from './AddressType';
  import { AddressTypeService } from '../maintenance/address-type-maintenance/address-type.service';

  export class AddressTypeActionResult {
    success = false;
    statusMessages: string[] = [];
    addressType: AddressType = null;

    constructor(private addressTypeService: AddressTypeService) {
      this.success = false;
      this.addressType = this.addressTypeService.newAddressType();
      this.statusMessages = [];
    }
  }
